combat-trainer.lic

```diff
@@ -189,12 +190,43 @@ class LootProcess
   def dispose_body(game_state)
     return if DRRoom.dead_npcs.length < 1
     game_state['mob_died'] = true
-
-    check_skinning(DRRoom.dead_npcs.first) if @skin
+    
+#NECRO THANATOLOGY CHECK STUFF GOES BELOW
+    
+    if @necro {check_thanatology(DRRoom.dead_npcs.first)}
+      dispose_body_loot
+      
+#NECRO THANATOLOGY CHECK STUFF GOES ABOVE
+    
+    elseif @skin { check_skinning(DRRoom.dead_npcs.first) }
+      dispose_body_loot
 
     bput('loot', 'You search', 'I could not find what you were referring to')
     pause 0.2 while DRRoom.dead_npcs.length > 0
   end
+  
+#NECRO THANATOLOGY RITUAL METHOD GOES BELOW
+  
+  def check_thanatology(mob_noun)
```

Comment:
every time you move into a new block of code (def, when, if...) you need a corresponding end to close that block.